.github/workflows/pyright.yml

```diff
@@ -34,6 +34,24 @@ jobs:
                   #    See https://anaconda.org/anaconda/python for available "python" versions.
                   python-version: ${{ matrix.python-version }}
 
+            - name: Load cached env
+              uses: actions/cache/restore@v4
+              id: cache-miniconda-load
+              with:
+                  path: /usr/share/miniconda/envs/in-context-learning
+                  key: ${{ runner.os }}-${{ hashFiles('environment.yaml') }}
+
+            - name: Update the conda env
+              if: steps.cache-miniconda-load.outputs.cache-hit != 'true'
+              run: conda env update -n in-context-learning -f environment.yaml
```

Comment:
if the conda environment exists and was saved to cache, but was changed in a commit, will this skip updating the conda environment? if not, this PR seems ready to merge